src/lib/link.ts

```diff
@@ -152,44 +152,83 @@ export class Link {
   // TODO: define ordering type
   /* eslint @typescript-eslint/no-unused-vars: "off" */
   public async createChannel(
-    _sender: Side,
-    _srcPort: string,
-    _destPort: string,
-    _order: string,
-    _version: string
+    sender: Side,
+    srcPort: string,
+    destPort: string,
+    ordering: Order,
+    version: string
   ): Promise<ChannelPair> {
-    throw new Error('unimplemented');
-  }
-
-  // TODO: relayAllPendingPackets (filter)
-  // TODO: relayAllPendingAcks (filter)
-  // TODO: relayAllRoundTrip (filter)
-  // TODO: relayRoundTrip (packet)
+    const { src, dest } = this.getEnds(sender);
 
-  //   // CreateChannel constructs and executes channel handshake messages in order to create
-  // // OPEN channels on chainA and chainB. The function expects the channels to be successfully
-  // // opened otherwise testing will fail.
-  // func (coord *Coordinator) CreateChannel(
-  // 	chainA, chainB *TestChain,
-  // 	connA, connB *ibctesting.TestConnection,
-  // 	sourcePortID, counterpartyPortID string,
-  // 	order channeltypes.Order,
-  // ) (ibctesting.TestChannel, ibctesting.TestChannel) {
+    // init on src/A
```

Comment:
How is it that can `src` and `A` be equated? Shouldn't this just be `src`?

Reply:
right you are